base/dcp_client_test.go

```diff
@@ -385,3 +385,26 @@ func TestResumeStoppedFeed(t *testing.T) {
 	}
 
 }
+
+// TestBadAgentPriority makes sure we can not specify agent priority as high
+func TestBadAgentPriority(t *testing.T) {
+	if UnitTestUrlIsWalrus() {
+		t.Skip("This test only works against Couchbase Server, since DCPClient requires collections")
+	}
+
+	bucket := GetTestBucket(t)
+	defer bucket.Close()
+
+	feedID := "fakeID"
+	panicCallback := func(event sgbucket.FeedEvent) bool {
+		panic("Should not hit this callback")
+	}
+	dcpClientOpts := DCPClientOptions{
+		AgentPriority: gocbcore.DcpAgentPriorityHigh,
+	}
+	collection, err := AsCollection(bucket)
+	require.NoError(t, err)
+	dcpClient, err := NewDCPClient(feedID, panicCallback, dcpClientOpts, collection)
+	require.Error(t, err)
+	require.Nil(t, dcpClient)
```

Comment:
If we don't get an error here - do we want the the test to close the `DCPClient` we managed to erroneously open? I think we'd leak it otherwise.

Reply:
calling NewDCPClient doesn't actually open the dcp client, it just sets it up! Start() is what actually opens it.